services/errors.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+class NotFound extends Error {
+  constructor(message) {
+    super(message || 'not found');
+    this.name = 'NotFound';
+  }
+}
+
+class InvalidResponse extends Error {
+  constructor(message) {
+    super(message || 'invalid');
+    this.name = 'InvalidResponse';
+  }
+}
+
```

Comment:
I'm really confused about where the 'inaccessible'/connection error gets thrown. It does seem to work if I add a test with `.networkOff()` but I just can't find where it happens.

Reply:
Huh, I've just tried adding a test with `.networkOff()` but I can't get it to work. Not only that, the problem spills over into subsequent tests.

I wonder if this is a bug in our error handling, as a result of mixing promises and callbacks somewhere. AFAIK the exception raised by `.networkOff()` bubbles through the `err` to the request callback which means it should definitely not be unhandled.

Although… our [new handler function](https://github.com/badges/shields/blob/ea4b758612c6e87ba227773fddee7ceab03ee138/services/base.js#L144-L155) is an `async` method. It returns a promise, which means when it errors out, it's a rejected promise. However, [the callsite](https://github.com/badges/shields/blob/ea4b758612c6e87ba227773fddee7ceab03ee138/lib/request-handler.js#L183) doesn't check if the function returns anything, nor `.catch()` it. Probably the return value of that function invocation should be checked, and if it's a promise, a `catch()` should be chained onto it.

Reply:
Right sorry. I had overlooked that `appveyor.tester.js` has tests for the `/ci` endpoint (which has been migrated to the new architecture) and the `/tests` endpoint (which is still implemented in `server.js`). I added a test for the `/tests` endpoint. No wonder I was getting confused :D If I add a test for the right endpoint I get the same behaviour you've described.

I guess then the comment here is: we should also handle the 'connection error' case in the standard error handling, maintaining as much consistency with `NotFound` and `InvalidResponse` as possible (rule of least surprise and all that), accepting the caveat you've mentioned above.